--- ./fastrtps/src/cpp/rtps/transport/TCPChannelResource.cpp	2024-06-28 09:49:17.000000000 +0000
+++ ./fastrtps/src/cpp/rtps/transport/TCPChannelResource.cpp	2024-06-28 11:04:46.773302403 +0000
@@ -372,7 +372,7 @@
 }
 
 void TCPChannelResource::set_socket_options(
-        asio::basic_socket<asio::ip::tcp>& socket,
+        asio::basic_socket<asio::ip::tcp, asio::stream_socket_service<asio::ip::tcp> >& socket,
         const TCPTransportDescriptor* options)
 {
     uint32_t minimum_value = options->maxMessageSize;
